themes/bootstrap3/templates/record/cover.phtml

```diff
@@ -5,5 +5,5 @@
   <img src="<?=$this->escapeHtmlAttr($cover); ?>" <?php if ($linkPreview): ?>data-linkpreview="true" <?php endif; ?>class="recordcover" alt="<?=$alt; ?>" />
   <?php if ($this->link && !empty($alt)): ?></a><?php endif; ?>
 <?php else: ?>
-  <img src="<?=$this->url('cover-unavailable')?>" <?php if ($linkPreview): ?>data-linkpreview="true" <?php endif; ?>class="recordcover" alt="" />
+  <img src="<?=$this->url('cover-unavailable')?>" <?php if ($linkPreview): ?>data-linkpreview="true" <?php endif; ?>data-recordsource="<?=$driver->getSourceIdentifier()?>" data-recordid="<?=$driver->getUniqueID()?>" data-coversize="<?=$size?>" class="recordcover" alt="" />
```

Comment:
It's a little disconcerting that images display as unavailable before popping in with AJAX. Do we need to differentiate between "loading" status and "unavailable" status? (If that's too much hassle to deal with right now, I can live with it as-is in the interest of getting the feature merged, but I think we should at least open a JIRA ticket to make that a future enhancement to the system).

Reply:
I implemented a simple spinner here now

Reply:
Thanks! It wasn't quite working for me; for some reason, the paths in compiled.css were wrong. I ran `grunt less` again and it fixed everything. How are you compiling your LESS? I wonder if there is a bug in the process somewhere, or some inconsistency between our environments.

I also changed the text to simply "Loading" so we can use existing translations without having to create a new cover-specific one.

Reply:
I am compiling into CSS using util/cssBuilder command in Vagrant box. Then it worked for me on the same box as server.

Reply:
I think that the PHP compiler sometimes produces less portable code than the Javascript-based compiler. I'd suggest using `grunt less` instead of util/cssBuilder if practical... but if that's not possible, it's not a problem for me to recompile on my end as needed. :-)

Reply:
thanks @demiankatz I'll try it next time ;)